### PR TITLE
Ft csvPatch

### PR DESCRIPTION
**Pull request by summercode18**
Fixes # VORTEX-6247:Reordering CSV columns

## Proposed Changes
Bug caused by duplicates of the Index and Color columns which must always exist.
  - When user has these fields automatically re-name on import to prevent memory confusion.
  - Retains user color & potential custom indexing.
---Could be modified to override our Index + Color with user; but may not neccessarily make sense to do so.

### CODE REVIEW

File: open-sphere-plugins/csv-common/src/main/java/io/opensphere/csvcommon/ui/controller/ImportableColumnController.java [hunk 1]
```diff
@@ -46,6 +49,22 @@ public void detectNonImportableColumns(CSVParseParameters parameters, Preference
...
 
+                List<?> currentParams = new ArrayList<String>();
+                currentParams = parameters.getColumnNames();
+                List<String> newParams = new ArrayList<String>();
+
+                Iterator<?> litr = currentParams.iterator();
+                while (litr.hasNext())
+                {
+                    Object element = litr.next();
+                    if (element.equals("Index") || element.equals("Color"))
```
Comment (brelsftm) [inline open-sphere-plugins/csv-common/src/main/java/io/opensphere/csvcommon/ui/controller/ImportableColumnController.java]: Im not sure harding the column name is a good idea in this case. Ill have to take look at the ticket

File: open-sphere-plugins/csv-common/src/main/java/io/opensphere/csvcommon/ui/controller/ImportableColumnController.java [hunk 2]
```diff
@@ -1,6 +1,9 @@
 package io.opensphere.csvcommon.ui.controller;
 
+import java.util.ArrayList;
+import java.util.Iterator;
```
Comment (brelsftm) [inline open-sphere-plugins/csv-common/src/main/java/io/opensphere/csvcommon/ui/controller/ImportableColumnController.java]: is Iterator used anywhere? Does this import need to be here

File: open-sphere-plugins/csv-common/src/main/java/io/opensphere/csvcommon/ui/controller/ImportableColumnController.java [hunk 3]
```diff
@@ -1,6 +1,9 @@
 package io.opensphere.csvcommon.ui.controller;
 
+import java.util.ArrayList;
+import java.util.Iterator;
 import java.util.List;
+import java.util.ListIterator;
```
Comment (brelsftm) [inline open-sphere-plugins/csv-common/src/main/java/io/opensphere/csvcommon/ui/controller/ImportableColumnController.java]: Does ListIterator need to be imported?

File: open-sphere-plugins/csv-common/src/main/java/io/opensphere/csvcommon/ui/controller/ImportableColumnController.java [hunk 4]
```diff
@@ -46,6 +49,22 @@ public void detectNonImportableColumns(CSVParseParameters parameters, Preference
 
                 List<String> columnsToNotImport = CSVColumnPrefsUtil.getCustomKeys(registry, key);
 
+                List<?> currentParams = new ArrayList<String>();
```
Comment (brelsftm) [inline open-sphere-plugins/csv-common/src/main/java/io/opensphere/csvcommon/ui/controller/ImportableColumnController.java]: List<String> currentParams =  parameters.getColumnNames();

File: open-sphere-plugins/csv-common/src/main/java/io/opensphere/csvcommon/ui/controller/ImportableColumnController.java [hunk 5]
```diff
@@ -46,6 +48,22 @@ public void detectNonImportableColumns(CSVParseParameters parameters, Preference
 
                 List<String> columnsToNotImport = CSVColumnPrefsUtil.getCustomKeys(registry, key);
 
+                List<?> currentParams = new ArrayList<String>();
+                currentParams = parameters.getColumnNames();
+                List<String> newParams = new ArrayList<String>();
+
+                Iterator<?> litr = currentParams.iterator();
+                while (litr.hasNext())
```
Comment (Xenotide) [inline open-sphere-plugins/csv-common/src/main/java/io/opensphere/csvcommon/ui/controller/ImportableColumnController.java]: Could replace this while iterator/while loop with currentParams.forEach()

File: open-sphere-plugins/csv-common/src/main/java/io/opensphere/csvcommon/ui/controller/ImportableColumnController.java (hunk 3 above)
Comment (summercode18) [reply open-sphere-plugins/csv-common/src/main/java/io/opensphere/csvcommon/ui/controller/ImportableColumnController.java]: removed

**Comment by brelsftm** [comment]
Ill leave it up to matt to approve this when its ready since you guys have been working on it

File: open-sphere-plugins/csv-common/src/main/java/io/opensphere/csvcommon/ui/controller/ImportableColumnController.java (hunk 1 above)
Comment (summercode18) [reply open-sphere-plugins/csv-common/src/main/java/io/opensphere/csvcommon/ui/controller/ImportableColumnController.java]: replaced with checking all metacolumn cases

File: open-sphere-plugins/csv-common/src/main/java/io/opensphere/csvcommon/ui/controller/ImportableColumnController.java (hunk 4 above)
Comment (summercode18) [reply open-sphere-plugins/csv-common/src/main/java/io/opensphere/csvcommon/ui/controller/ImportableColumnController.java]: added

File: open-sphere-plugins/csv-common/src/main/java/io/opensphere/csvcommon/ui/controller/ImportableColumnController.java (hunk 5 above)
Comment (summercode18) [reply open-sphere-plugins/csv-common/src/main/java/io/opensphere/csvcommon/ui/controller/ImportableColumnController.java]: replaced

**Pull request merged**